pairing/adapter.go

```diff
@@ -0,0 +1,47 @@
+package pairing
+
+import (
+	"github.com/dedis/kyber"
+	"github.com/dedis/kyber/pairing/bn256"
+)
+
+// SuiteBn256 is an adapter from the different bn256 suites to a single one
+// that can be used by onet to encode/decode key pairs. The public key comes
+// from the field G2 and the private key comes from G1 thus the Point and
+// Scalar functions return the corresponding type.
```

Comment:
I might be wrong, but I suspect that it doesn't matter which group the pk is from. Out BLS implementation I think is similar to this one https://en.wikipedia.org/wiki/Boneh%E2%80%93Lynn%E2%80%93Shacham where the pk (g^x) is in G2. But for this one https://crypto.stanford.edu/~dabo/pubs/papers/BLSmultisig.html the pk is in G1 (G0 in their notation). I think as long as you run the other pairing operation correctly then it's fine. G1 and G2 have the same order so I think you can sign a message in both groups as long as you verify it correctly. Also, the scalar has the same order in both groups so it doesn't matter where it comes from.

Maybe there's nothing to be done here but update the comment to say this adapter is specific to our BLS implementation.

Reply:
Actually it's not similar because we have different groups G_1 and G_2 with respectively GF(p) and GF(p^2). It's true the scalar has the same order though. The pk is in G2 (G1 for them) in the Standford paper and the signature in G1 (G0 for them).

I will add the comment.